### ISSUE TITLE
AttributeError on bad skip comment

### ISSUE THREAD

**Issue by adamtheturtle** [simplistix/sybil#140]
Add this comment to a rST file:

```python
.. skip<:
```

and use the `sybil.parsers.rest.skip import SkipParser`, and you will see:

```
../../../.virtualenvs/vws-python/lib/python3.12/site-packages/sybil/document.py:53: in parse
    for region in parser(document):
../../../.virtualenvs/vws-python/lib/python3.12/site-packages/sybil/parsers/abstract/skip.py:28: in __call__
    yield Region(lexed.start, lexed.end, match.groups(), self.skipper)
E   AttributeError: 'NoneType' object has no attribute 'groups'
```

**Comment by cjw296** [comment]
Interesting :-) Fancy working up a PR to give a better error message?

**Comment by adamtheturtle** [comment]
I'm not against contributing but I'm not sure I can. First off, I'd like to have a shared idea of a goal. It probably would be no error message - my thinking is that this comment is not a Sybil comment / a comment Sybil should be looking for at all, so there should be no error (and nothing should be skipped).

**Comment by cjw296** [comment]
Did you hit this intentionally or by a typo?

**Comment by adamtheturtle** [comment]
On the topic we were discussing before, I was trying to add custom skips for particular Sybils. I wanted a general skip of `.. doccmd skip: next` and a more specific skip of e.g. `.. doccmd skip[type-checking]: next`. I hit this issue while developing that, and then narrowed down the problem to what I have described in this issue.

**Comment by adamtheturtle** [comment]
Even if it were a typo, Sybil doesn't pick up on `.. skop: next` (untested), and I don't think it should.

**Comment by cjw296** [comment]
There is a balance here: if something is structurally valid but omitted because it is of a directive type that we're not interested, then sure, no error.

If it's not a valid directive, or otherwise contains an error that the author is likely to have made by mistake, then we probably want to let them know. Sure, it'll become apparent soon enough that a directive has not been executed, but in cases where we can give a helpful exception sooner, I think Sybil should.

In this case its more nuanced, as skip and a couple of other Sybil things use pseudo-directives so as not to cause errors and warnings with other tools that process this text, particularly Sphinx.

The exception you originally mentions should be saying to the user "hey, I found a skip directive, but I couldn't parse it's arguments"

**Comment by adamtheturtle** [comment]
I believe that if this remains an error, it is impossible in Sybil to build have two skip decorators, one `.. doccmd skip: next` and one `.. doccmd skip[type-checking]`. For my purposes, I've gone with `.. doccmd skip[all]` and `.. doccmd skip[type-checking]`, but this limitation of the parsing logic was not expected by me.

**Comment by cjw296** [comment]
The thing I'd say here: would `.. doccmd skip[type-checking]::` be a valid ReST directive? 
If so, then Sybil should support the same structure as a pseudo-directive.

If not, then the above exception could be polished to be more helpful.

Any reason not to use `.. doccmd-skip-all` and `.. doccmd-skip-typing`? (suggesting those as they seem obvious valid in terms of being ReST directives)

**Comment by cjw296** [comment]
> I think what you're saying is that this is looking for something that is like a valid ReST directive but with only one : instead of two.

Correct.

> My intention is that, like with skip: next, doccmd skip works the same across Markdown, ReST, MyST (and whatever comes next... djot, ...?).

Have a look at how skip works in MyST and Markdown - my intention is to use whatever is closest to a real directive in whatever markup is being used. I'm not sure "works the same across Markdown, ReST, MyST (and whatever comes next..." is reasonable, TBH, it depends on what works in those. If you're looking for a common denominator, then it'll want to be simple and likely use as little punctuation as possible.

>  Please let me know if we're going too far into my personal use case!

It may be ;-)

**Comment by adamtheturtle** [comment]
Thanks for the response. I think it is best in this case that I leave this as a bug report and for you to choose what's best for Sybil.

**Comment by cjw296** [comment]
I'm actually a little surprised and disappointing that mypy doesn't complain about this.
Surely we should have had a type checking failure saying basically what the error above says?

**Comment by cjw296** [comment]
Addressed in 39bf168ca1f1e2a43e29fb7aad50dfbf5e9763ad.

**Comment by adamtheturtle** [comment]
> I'm actually a little surprised and disappointing that mypy doesn't complain about this.
> Surely we should have had a type checking failure saying basically what the error above says?

@cjw296 In `mypy.ini` this error code is disabled`:

```ini
disable_error_code = union-attr,return
```

If `union-attr` is not disabled, we have this error:

```
sybil/parsers/abstract/skip.py:28: error: Item "None" of "Match[str] | None" has no attribute "groups"  [union-attr]
```

As an aside, the are three remaining `union-attr` errors after your fix.
